### PR TITLE
SWC-6549 - Add support for viewing and querying JSON Subcolumn Facets. Refactor existing facet components.

### PR DESCRIPTION
**Pull request by nickgros**
- JsonColumnFacetFilters: New component that renders a set of JSON Subcolumn facets using specific styling to match DESIGN-1281
- FacetFilterControls: Group JSON Subcolumn facets by column and pass to JsonColumnFacetFilters
- Update FacetNav to correctly plot and label JSON Subcolumn facets
- Refactor useImmutableTableQuery to use a reducer, preventing cases where state updates would be improperly batched and dropped
- Change Enum/RangeFacetFilter prop signatures to no longer require column data, which can be found from the query data
- Refactor EnumFacetFilter - create new subcomponents to support adding UI tests without requiring a full QueryWrapper setup.
- Refactor RangeFacetFilter/CombinedRangeFacetFilter to use same UI base component
- Change prop names in Range components for clarity
- Drop unused/invariant props across many facet components

### CODE REVIEW

File: packages/synapse-react-client/demo/containers/playground/WidgetDemo.tsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ export const WidgetDemo: React.FunctionComponent = () => {
       <Range
         type="number"
         initialValues={{ min: rangeNumberValue.min, max: rangeNumberValue.max }}
-        onChange={(values: RangeValues) => setRangeNumberValue(values)}
+        onApplyClicked={(values: RangeValues) => setRangeNumberValue(values)}
```
Comment (nickgros) [inline packages/synapse-react-client/demo/containers/playground/WidgetDemo.tsx:128]: Changed some of these prop names for clarity

File: packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx [hunk 2]
```diff
@@ -46,6 +46,7 @@ export type QueryContextType<
   /** Resets the query to its initial state, clearing all filters added by the user */
   resetQuery: ImmutableTableQueryResult['resetQuery']
   addValueToSelectedFacet: ImmutableTableQueryResult['addValueToSelectedFacet']
+  setRangeFacetValue: ImmutableTableQueryResult['setRangeFacetValue']
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx:49]: Start to pull range facet controls into `QueryContext`

File: packages/synapse-react-client/src/components/SynapseTable/SynapseTableRenderers.tsx [hunk 3]
```diff
@@ -203,11 +203,7 @@ export function TableDataColumnHeader(
       <div className="SRC-centerContent" style={{ height: '22px' }}>
         {isFacetSelection && !isLockedColumn && columnModel && (
           <span>
-            <EnumFacetFilter
-              containerAs="Dropdown"
-              facetValues={facet.facetValues}
-              columnModel={columnModel}
-            />
+            <EnumFacetFilter containerAs="Dropdown" facet={facet} />
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/SynapseTable/SynapseTableRenderers.tsx:206]: Simplified `EnumFacetFilter` prop signature; just pass the facet, it will figure out the columnModel internally (so we don't have to do the same thing all over the place)

File: packages/synapse-react-client/src/components/widgets/RadioGroup.tsx [hunk 4]
```diff
@@ -46,7 +46,7 @@ export function RadioOption<T extends string | boolean | number = string>(
 ) {
   const [uniqueId] = useState(_uniqueId('src-radio-'))
   return (
-    <div onClick={() => props.onChange(props.value)}>
+    <div className={'radio'} onClick={() => props.onChange(props.value)}>
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/RadioGroup.tsx:49]: Fixes label styling

File: packages/synapse-react-client/src/components/widgets/Range.tsx [hunk 5]
```diff
@@ -13,13 +13,11 @@ export type RangeProps = {
   initialValues?: RangeValues
   className?: string
   errorText?: string
-  onChange: (newValues: RangeValues) => void
+  onApplyClicked: (newValues: RangeValues) => void
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/Range.tsx:16]: `onApplyClicked` is more clear since this component contains two input fields that the user can update. Hopefully it is now obvious that this callback only is invoked when the user clicks "Apply"

File: packages/synapse-react-client/src/components/widgets/RangeSlider/RangeSlider.tsx [hunk 6]
```diff
@@ -14,8 +14,8 @@ export type RangeSliderProps = React.PropsWithChildren<{
   domain: string[]
   initialValues: RangeValues
   step: number
-  doUpdateOnApply?: boolean
-  onChange: (values: RangeValues) => void
+  onChange?: (values: RangeValues) => void
+  onApplyClicked?: (values: RangeValues) => void
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/RangeSlider/RangeSlider.tsx:18]: Removed `doUpdateOnApply`, which decided whether `onChange` should be invoked "on change" or "on apply clicked".

Instead, add `onApplyClicked`. Always call each callback when the user does what the callback describes.

File: packages/synapse-react-client/src/components/widgets/facet-nav/SelectionCriteriaPills.tsx [hunk 7]
```diff
@@ -135,6 +139,20 @@ function getPillPropsFromQueryFilters(
...
+    return `Up to ${max}`
+  } else if (max == undefined) {
+    return `${min} or greater`
+  } else if (min === VALUE_NOT_SET && max === VALUE_NOT_SET) {
+    return FRIENDLY_VALUE_NOT_SET
+  } else {
+    return `${min} - ${max}`
+  }
+}
+
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/facet-nav/SelectionCriteriaPills.tsx:155]: Handle this bug 

image

File: packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.integration.test.tsx [hunk 8]
```diff
@@ -18,27 +18,22 @@ import {
...
+let capturedOnApplyClicked:
   | ((range: { min: string | number; max: string | number }) => void)
   | undefined
 
-jest.mock('../Range', () => ({
-  Range: jest.fn((props: RangeProps) => {
-    capturedOnChange = props.onChange
-    return <div data-testid="Range"></div>
-  }),
-}))
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.integration.test.tsx:38]: Don't mock the `Range` input field component

File: packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.tsx [hunk 9]
```diff
@@ -8,17 +8,17 @@ import { Range, RangeValues } from '../Range'
...
 import { SynapseConstants } from '../../../utils'
 import { useGetQueryResultBundleWithAsyncStatus } from '../../../synapse-queries'
+import { useQueryVisualizationContext } from '../../QueryVisualizationWrapper'
 
 export type CombinedRangeFacetFilterProps = {
   facetResults: FacetColumnResultRange[]
   label: string
   columnType: ColumnType
   onChange: (range: (RadioValuesEnum | number | string | undefined)[]) => void
-  collapsed?: boolean
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.tsx:21]: We don't ever set `collapsed` state to `true`, so remove the prop from all of the facet control components

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter.stories.tsx
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter.stories.tsx:1]: All of the EnumFacetFilter components were moved to an EnumFacetFilter folder. git seems to think they are new files

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.integration.test.tsx
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.integration.test.tsx:1]: No "new" tests, but was updated to just test the subset of logic (mostly wrangling facets) that remains in `EnumFacetFilter.tsx`

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import {
+  FRIENDLY_VALUE_NOT_SET,
+  VALUE_NOT_SET,
+} from '../../../../utils/SynapseConstants'
+import { cloneDeep } from 'lodash-es'
+import EnumFacetFilterUI, { RenderedFacetValue } from './EnumFacetFilterUI'
+
+const meta = {
+  title: 'Explore/Components/Facets/EnumFacetFilter',
+  component: EnumFacetFilterUI,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.stories.tsx:12]: Note that this story now inspects the `EnumFacetFilterUI` component. This component does not rely on QueryWrapper, making it a better candidate for a story.

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+import { getAllIsSelected, valueToLabel } from './EnumFacetFilterUtils'
+
+export type EnumFacetFilterProps = {
+  facet: FacetColumnResultValues
+  containerAs?: 'Collapsible' | 'Dropdown'
+  dropdownType?: 'Icon' | 'SelectBox'
+  hideCollapsible?: boolean
+}
+
+export function EnumFacetFilter(props: EnumFacetFilterProps) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx:34]: Component just assembles all of the data needed to render `EnumFacetFilterUI`

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+      ),
+      fv => -1 * fv.count,
+    )
+  }, [
+    currentSelectedFacet?.facetValues,
+    entityHeaders,
+    evaluations,
+    facet.facetValues,
+    userGroupHeaders,
+  ])
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx:140]: Transform all of our facet data into a list of a custom type `RenderedFacetValue`, which is what EnumFacetFilterUI uses for display

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+      onRemoveValueFromSelection={value => {
+        removeValueFromSelectedFacet(
+          pick(facet, ['columnName', 'jsonPath']),
+          value,
+          { debounce: true },
+        )
+      }}
+      onRemoveAllFacetSelections={() =>
+        removeSelectedFacet(pick(facet, ['columnName', 'jsonPath']))
+      }
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx:170]: Also transform these component-specific callbacks to the actions provided by QueryWrapper

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterOption.tsx
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterOption.tsx:1]: This was already a separate component, I just moved it to its own file.

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+  onRemoveValueFromSelection: (value: string) => void
+  /* Called when all values should be unselected (e.g. 'All' is clicked) */
+  onRemoveAllFacetSelections: () => void
+}
+
+function removeWhitespace(value: string): string {
+  return value.replace(/\s/g, '').toLowerCase()
+}
+
+export default function EnumFacetFilterUI(props: EnumFacetFilterUIProps) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx:42]: Given a list of `RenderedFacetValue` (and all of the other props above), render the UI for the EnumFacetFilter. Most importantly, this is decoupled from `QueryWrapper`.

Some actions that this handles include
- Collapse/expand
- Truncating/expanding the list of facets to show
- Client-side search of facet values.

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+  const [showSearch, setShowSearch] = useState<boolean>(false)
+  const [searchTerm, setSearchText] = useState<string>('')
+
+  // e.g. we show a subset of facet values before hiding the rest behind "Show More"
+  // If the user has selected a value that would normally be hidden, always show all of the values
+  const valueIndexedGreaterThanMaxToShowIsSelected =
+    facetValues.length > MAX_ENUMERATION_VALUES_TO_SHOW &&
+    facetValues
+      .slice(MAX_ENUMERATION_VALUES_TO_SHOW)
+      .some(({ isSelected }) => isSelected)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx:65]: This is kind of strange, but it was the existing behavior. I suppose it makes sense to show you the full list if someone links you to a query where the 10th facet value is selected

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+    return facetValues.slice(0, MAX_ENUMERATION_VALUES_TO_SHOW)
+  }, [facetValues, searchTerm, showSearch, showAll])
+
+  const textInput: React.RefObject<HTMLInputElement> = React.createRef()
+
+  const isDropdown = containerAs === 'Dropdown'
+  const content = (
+    <div className={isDropdown ? 'EnumFacetFilter__dropdown_menu' : ''}>
+      <div className="EnumFacetFilter__checkboxContainer--forAll">
+        <TextField // Search Filter Text
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx:90]: Changed to use MUI

image

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+                  <div className="EnumFacetFilter__checkboxContainer">
+                    <div className="EnumFacetFilter__showMoreFacetsLabel">
+                      Show all ({facetValues.length})
+                    </div>
+                  </div>
+                </button>
+              )}
+            {showAll &&
+              !searchTerm &&
+              !valueIndexedGreaterThanMaxToShowIsSelected &&
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx:217]: Fixed an issue where
1. `showAll` is true because the e.g. 6th item is selected
2. The "Show less" button was shown
3. When clicked, "Show less" would do nothing

Now, hide the button if `valueIndexedGreaterThanMaxToShowIsSelected` is true.

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUtils.ts [hunk 18]
```diff
@@ -0,0 +1,66 @@
+import {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUtils.ts:1]: Extracted and tested these utils, didn't change substantially

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetChip.tsx [hunk 19]
```diff
@@ -1,23 +1,16 @@
...
-}
+}>
 
-export const FacetChip: React.FC<FacetChipProps> = ({
-  facet,
-  isChecked,
-  onClick,
-}) => {
-  const { getColumnDisplayName } = useQueryVisualizationContext()
+export function FacetChip(props: FacetChipProps) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetChip.tsx:9]: Changed no longer rely on the query context. Instead of getting a label from context, it just renders whatever you pass as a child.

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.integration.test.tsx [hunk 20]
```diff
@@ -145,41 +106,6 @@ describe('FacetFilterControls tests', () => {
   })
 
   describe('handling child component callbacks', () => {
-    it('should propagate range correctly', async () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.integration.test.tsx:148]: This logic was pulled out of this component, so it doesn't need to be tested here.

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx [hunk 21]
```diff
@@ -109,23 +119,6 @@ export function applyChangesToValuesColumn(
...
-) => {
-  facet.columnMin = values[0]
-  facet.columnMax = values[1]
-  const changedFacet = convertFacetColumnRangeRequest(facet)
-  const result = patchRequestFacets(
-    changedFacet,
-    lastRequest?.query?.selectedFacets,
-  )
-  onChangeFn(result)
-}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx:127]: Now handled by `useImmutableTableQuery` and the callback is provided by the QueryContext

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx [hunk 22]
```diff
@@ -149,21 +142,18 @@ export const applyCombinedChangesToRangeColumn = (
...
  * Determines which facet filters should be shown after loading a new bundle. The shown facets will be the first
  * three available facets, plus any other facets that have a filter applied.
- * @param facets
+ * @param facetColumns
  * @param selectedFacets
  * @returns the columnNames of the facets that should be shown.
  */
 export function getDefaultShownFacetFilters(
-  facets: FacetColumnResult[],
+  facetColumns: string[],
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx:150]: Since one column may now have multiple facets, the show/hide facet control functionality now operates on columns rather than facets

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterHeader.tsx [hunk 23]
```diff
@@ -1,33 +1,31 @@
...
+  hideCollapsible?: boolean
   isCollapsed: boolean
   onClick: (newValue: boolean) => void
 }
 
-export const FacetFilterHeader: React.FunctionComponent<
-  FacetFilterHeaderProps
-> = ({ label, isCollapsed, onClick }: FacetFilterHeaderProps) => {
-  const { getColumnDisplayName } = useQueryVisualizationContext()
+export function FacetFilterHeader(props: FacetFilterHeaderProps) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterHeader.tsx:12]: No longer relies on QueryContext. Just pass the name in the `label` prop

File: packages/synapse-react-client/src/components/widgets/query-filter/JsonColumnFacetFilters.tsx [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+  /* The set of JSON Subcolumn FacetColumnResults that will be shown in this panel */
+  facets: FacetColumnResult[]
+}
+
+/**
+ * Renders a set of facet filters that are all derived from the same JSON Column.
+ * @param props
+ * @constructor
+ */
+export default function JsonColumnFacetFilters(
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/JsonColumnFacetFilters.tsx:24]: This component lets us logically group a set of JSON subcolumn facets that are all derived from the same column

Comment (nickgros) [reply packages/synapse-react-client/src/components/widgets/query-filter/JsonColumnFacetFilters.tsx:24]: image

File: packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.integration.test.tsx
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.integration.test.tsx:1]: Not a new file, but git seems to think so.

Changed the tests to account for refactoring. No new tests, but test logic was changed so that
- Range component is no longer mocked
- Changes to the query are verified by monitoring the query context.

File: packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.tsx [hunk 25]
```diff
@@ -60,16 +63,29 @@ export const RangeFacetFilter: React.FunctionComponent<
...
+            selectedFacet =>
+              selectedFacet.columnName === facetResult.columnName &&
+              selectedFacet.jsonPath === facetResult.jsonPath,
+          )
+        if (selectedFacet) {
+          removeSelectedFacet(selectedFacet)
+        }
+
+        break
+      }
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.tsx:88]: Directly change the query using callbacks provided by `QueryContext` to replace custom callbacks that were previously defined in `FacetFilterControls.tsx` and passed in

File: packages/synapse-react-client/src/components/widgets/Range.tsx [hunk 26]
```diff
@@ -13,13 +13,11 @@ export type RangeProps = {
...
 }
 
-export const Range: React.FunctionComponent<RangeProps> = (
-  props: RangeProps,
-) => {
-  const errorText = 'Min value should be less then max value'
+export function Range(props: RangeProps) {
+  const errorText = 'Min value should be less than max value'
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/Range.tsx:20]: then -> than

File: packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.test.tsx [hunk 27]
```diff
@@ -194,7 +194,13 @@ describe('QueryWrapper', () => {
...
 
       act(() => {
-        providedContext!.executeQueryRequest(initialQueryRequest)
+        providedContext!.executeQueryRequest({
+          ...initialQueryRequest,
+          query: {
+            ...initialQueryRequest.query,
+            offset: 10,
+          },
+        })
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.test.tsx:203]: Previously tested `onQueryChange`, but the query didn't actually change. Update the test to actually change the query. Should we define/test the behavior for the edge case where the query was updated but not changed?

File: packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.tsx [hunk 28]
```diff
@@ -111,6 +111,7 @@ function _QueryWrapper(props: QueryWrapperProps) {
     nextQueryRequest,
     currentQueryRequest,
     addValueToSelectedFacet,
+    setRangeFacetValue,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.tsx:114]: Add to query wrapper to support updating range facets thru query context (just like enum facets)

File: packages/synapse-react-client/src/components/SynapseTable/SynapseTable.integration.test.tsx [hunk 29]
```diff
@@ -140,6 +143,8 @@ jest.spyOn(AddToDownloadListV2Module, 'default').mockImplementation(() => {
 describe('SynapseTable tests', () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/SynapseTable/SynapseTable.integration.test.tsx:143]: Minor changes in mock data used + setup to fix new issues/improve flakiness

File: packages/synapse-react-client/src/components/widgets/Range.test.tsx [hunk 30]
```diff
@@ -64,7 +64,7 @@ describe('Range input test', () => {
     await updateValue(4.9, 2)
     expect(mockCallback).not.toHaveBeenCalled()
     const errorMessage = screen.getByText(
-      'Min value should be less then max value',
+      'Min value should be less than max value',
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/Range.test.tsx:67]: typo

File: packages/synapse-react-client/src/components/widgets/facet-nav/FacetNav.tsx [hunk 31]
```diff
@@ -30,7 +34,7 @@ export type FacetNavProps = {
 }
 
 type UiFacetState = {
-  name: string
+  facet: UniqueFacetIdentifier
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/facet-nav/FacetNav.tsx:37]: Facets are now defined using the 2-tuple (columnName, jsonPath), so we can no longer track UI state using just a column name. Use the `UniqueFacetIdentifier` record that stores this data

File: packages/synapse-react-client/src/components/widgets/facet-nav/SelectionCriteriaPills.tsx (hunk 7 above)
Comment (nickgros) [reply packages/synapse-react-client/src/components/widgets/facet-nav/SelectionCriteriaPills.tsx:155]: image

File: packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx [hunk 32]
```diff
@@ -234,120 +184,134 @@ function FacetFilterControls(props: FacetFilterControlsProps) {
...
-                      facet,
-                      applyChanges,
-                      values as string[],
-                    )
-                  }
-                ></RangeFacetFilter>
-              )}
-            </div>
+      {shownTopLevelFacets.map(facet => {
+        const columnModel = getCorrespondingColumnForFacet(facet, columnModels!)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/FacetFilterControls.tsx:273]: I think I can remove this

File: packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.tsx [hunk 33]
```diff
@@ -25,130 +25,51 @@ export const options = [
...
-              )}
-
-              {columnType === 'DATE' && (
-                <Range
-                  key="Range"
-                  initialValues={{
-                    // From the backend, selectedMin is a formatted date (like "2021-06-15"), but columnMin is a unix timestamp in millis (like "1624651794856")
-                    min: selectedMin ?? dayjs(parseInt(columnMin)).toString(),
-                    max: selectedMax ?? dayjs(parseInt(columnMax)).toString(),
-                  }}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.tsx:129]: Does this logic still exist? Does it need to (probably)?

File: packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilterUI.tsx [hunk 34]
```diff
@@ -0,0 +1,161 @@
...
+        onAnySelected()
+
+        break
+      }
+    }
+  }
+
+  useEffect(() => {
+    setRadioValue(getRadioValue(currentMin, hasAnyValue))
+  }, [currentMin, hasAnyValue])
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilterUI.tsx:91]: Why do we need this?

Comment (nickgros) [reply packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilterUI.tsx:91]: AAdd a comment

File: packages/synapse-react-client/src/utils/hooks/useImmutableTableQuery/useTableQueryReducer.ts [hunk 35]
```diff
@@ -0,0 +1,376 @@
...
+
+      const isConfirmingChange =
+        requireConfirmationOnChange &&
+        !queryRequestsHaveSameTotalResults(
+          currentQueryRequest.query,
+          nextQueryRequestClone.query,
+        )
+
+      if (
+        (isConfirmingChange &&
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/hooks/useImmutableTableQuery/useTableQueryReducer.ts:317]: Add comment to explain cases

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx [hunk 36]
```diff
@@ -0,0 +1,173 @@
...
+          facet,
+        )
+        return undefined
+      }
+      return facetColumnRequest
+    }, [facet, nextQueryRequest.query.selectedFacets])
+
+  // Must compare the known facet values to the "uncommitted" current query
+  const allIsSelected = getAllIsSelected(
+    cloneDeep(nextQueryRequest) as QueryBundleRequest,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx:75]: Consider changing to `uncommittedQueryRequest` everywhere

File: packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx [hunk 37]
```diff
@@ -0,0 +1,277 @@
...
+                  onRemoveValueFromSelection(value)
+                }
+              }}
+            />
+          )
+        })}
+        {!isDropdown && (
+          <>
+            {!showAll &&
+              facetValues.length > MAX_ENUMERATION_VALUES_TO_SHOW && (
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx:203]: hasShowAllButton = ...

File: packages/synapse-react-client/src/template_style/_form.scss [hunk 38]
```diff
@@ -16,7 +16,7 @@
 
 .checkbox,
 .radio {
-  padding-left: 6px;
+  padding-left: 0px;
```
Comment (nickgros) [inline packages/synapse-react-client/src/template_style/_form.scss:19]: So checkboxes can be left-aligned with their container

File: packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.test.tsx (hunk 27 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.test.tsx:203]: It's not the worst thing if the query is executed when the query is updated but unchanged, so I would not prioritize this unit test.

File: packages/synapse-react-client/src/template_style/_form.scss (hunk 38 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/template_style/_form.scss:19]: https://sagebionetworks.jira.com/browse/PORTALS-2732 may need to be revisited

Comment (nickgros) [reply packages/synapse-react-client/src/template_style/_form.scss:19]: Think I have a full solution here: [`b8a6171` (#475)](https://github.com/Sage-Bionetworks/synapse-web-monorepo/pull/475/commits/b8a6171c043ebb35dec9a4b8a0a9fefa096cc766)

**Review by jay-hodgson — APPROVED** [review]
